pkg/controller/controller_binding.go

```diff
@@ -555,6 +561,7 @@ func (c *controller) reconcileServiceBinding(binding *v1beta1.ServiceBinding) er
 		}
 
 		clearServiceBindingCurrentOperation(toUpdate)
+		toUpdate.Status.UnbindStatus = v1beta1.ServiceBindingUnbindStatusRequired
```

Comment:
Ok i moved the set to just above `brokerClient.Bind`, it looks like it is similar to what instance is doing: https://github.com/kubernetes-incubator/service-catalog/blob/cc816eba0eabdab6588b3da6a6b5f1b948e2cb3a/pkg/controller/controller_instance.go#L785 But I don't understand how this or instance capture the change on catastrophic errors since there is no call to `updateServiceBindingStatus`. 

Does this update look right?